### ISSUE TITLE
brocadevtm_pool resource cannot read nodes_list

### ISSUE THREAD

**Issue by peterbale** [sky-uk/terraform-provider-brocadevtm#77]
Seeing an issue where if you create a new pool using the 'brocadevtm_pool' resource and 'nodes_list' parameter it successfully creates it for you, however, when you want to change the pool 'nodes_list' it can't read the current loadbalancer state.

Below is the code snippet of the read function which doesn't contain the new 'nodes_list' key:
```
func resourcePoolRead(d *schema.ResourceData, m interface{}) error {

	config := m.(map[string]interface{})
	client := config["jsonClient"].(*api.Client)
	var poolName string

	if v, ok := d.GetOk("name"); ok {
		poolName = v.(string)
	}

	var poolObj pool.Pool
	client.WorkWithConfigurationResources()
	err := client.GetByName("pools", poolName, &poolObj)
	if err != nil {
		if client.StatusCode == http.StatusNotFound {
			d.SetId("")
			return nil
		}
		return fmt.Errorf("BrocadeVTM Pool error whilst retrieving %s: %s", poolName, err)
	}

	d.Set("name", poolName)
	d.Set("bandwidth_class", poolObj.Properties.Basic.BandwidthClass)
	d.Set("failure_pool", poolObj.Properties.Basic.FailurePool)
	d.Set("max_connection_attempts", *poolObj.Properties.Basic.MaxConnectionAttempts)
	d.Set("max_idle_connections_pernode", *poolObj.Properties.Basic.MaxIdleConnectionsPerNode)
	d.Set("max_timed_out_connection_attempts", *poolObj.Properties.Basic.MaxTimeoutConnectionAttempts)
	d.Set("monitors", poolObj.Properties.Basic.Monitors)
	d.Set("node_close_with_rst", *poolObj.Properties.Basic.NodeCloseWithReset)
	d.Set("node_connection_attempts", *poolObj.Properties.Basic.NodeConnectionAttempts)
	d.Set("node_delete_behaviour", poolObj.Properties.Basic.NodeDeleteBehavior)
	d.Set("node_drain_to_delete_timeout", *poolObj.Properties.Basic.NodeDrainDeleteTimeout)
	d.Set("nodes_table", poolObj.Properties.Basic.NodesTable)
	d.Set("note", poolObj.Properties.Basic.Note)
	d.Set("passive_monitoring", *poolObj.Properties.Basic.PassiveMonitoring)
	d.Set("persistence_class", poolObj.Properties.Basic.PersistenceClass)
	d.Set("transparent", *poolObj.Properties.Basic.Transparent)
	d.Set("auto_scaling", []pool.AutoScaling{poolObj.Properties.AutoScaling})
	d.Set("pool_connection", []pool.Connection{poolObj.Properties.Connection})
	d.Set("dns_autoscale", []pool.DNSAutoScale{poolObj.Properties.DNSAutoScale})
	d.Set("ftp", []pool.FTP{poolObj.Properties.FTP})
	d.Set("http", []pool.HTTP{poolObj.Properties.HTTP})
	d.Set("kerberos_protocol_transition", []pool.KerberosProtocolTransition{poolObj.Properties.KerberosProtocolTransition})
	d.Set("load_balancing", []pool.LoadBalancing{poolObj.Properties.LoadBalancing})
	d.Set("node", []pool.Node{poolObj.Properties.Node})
	d.Set("smtp", []pool.SMTP{poolObj.Properties.SMTP})
	d.Set("ssl", []pool.Ssl{poolObj.Properties.Ssl})
	d.Set("tcp", []pool.TCP{poolObj.Properties.TCP})
	d.Set("udp", []pool.UDP{poolObj.Properties.UDP})
	return nil
}
```

After manually tainting the resource we can see that terraform doesn't see the correct loadbalancer state despite the pool containing ["10.92.194.199:9070", "10.92.194.200:9070"]:

```
-/+ brocadevtm_pool.lb (tainted) (new resource required)
      id:                                "p0_lb_SSL" => <computed> (forces new resource)
      max_idle_connections_pernode:      "50" => <computed>
      max_timed_out_connection_attempts: "2" => <computed>
      monitors.#:                        "1" => "1"
      monitors.0:                        "Ping" => "Ping"
      name:                              "p0_lb_SSL" => "p0_lb_SSL"
      node_close_with_rst:               "false" => <computed>
      node_connection_attempts:          "3" => <computed>
      node_delete_behaviour:             "immediate" => "immediate"
      nodes_list.#:                      "2" => "2"
      nodes_list.1641969316:             "10.92.194.200:9090" => "10.92.194.200:9090"
      nodes_list.3531871385:             "10.92.194.199:9090" => "10.92.194.199:9090"
      nodes_table.#:                     "0" => <computed>
      passive_monitoring:                "true" => <computed>
      transparent:                       "false" => <computed>
```

**Comment by peterbale** [comment]
On closer inspection it seems to be the update method which isn't working as intended too. The provider acknowledges that a change needs to be made, 'makes it' and then believes this has been fixed. Subsequently when you run a plan on this it believes the changes has been successfully made.

**Comment by peterbale** [comment]
Fixed with #78
